### DISCUSSION TITLE
Esera Adapter and read only configuration

### DISCUSSION THREAD

**Discussion by thoro** [john30/ebusd#735] [General]
Hi,

I have an Esera Adapter, which seems to be broken since it can't write to the bus, I'm already on the list for the ebus 3 adapters, but would like to get some info out of the heating system regardless.

How can I setup ebusd so that it just interprets the messages received via the bus?

**Comment by mwildbolz** [comment]
You can remove the --scanconfig option and place just those configuration files in the config dir, which are suitable for your hardware config - for example templates, broadcast, vr_71, hmu - depending on your hardware configuration!

But there are only a few messages transferred regularly (e.g. Status, OutsideTemp, Time), for polling the rest you need write access to the bus.
I'm facing the same issue at the moment because my already repaired ESERA bus coupler is not able to write on the bus...

Good luck!

**Reply by thoro** [reply]
Do you by any chance know which files are required for following products:

VR71
VR921
VWW 11/4 SI
VWF 114/3
recovair 360/4 E

I'm interested in getting the groundwater temperatures, and maybe some details about the compressor.

The vr71 I included already and see the temperatures and status for the 3 heating circuits.

I have added already these files:

_templates.csv

vaillant:
08.hmu.csv
26.vr_71.csv
35.rcc.1.csv
broadcast.csv
errors.inc
hcmode.inc
_templates.csv

Not sure about the 35.rcc.1.csv - I have.

ebusd reports following masters:

address 00: master #1
address 03: master #11
address 08: slave #11
address 10: master #2
address 15: slave #2
address 26: slave
address 31: master #8, ebusd
address 33: master #13
address 36: slave #8, ebusd
address 38: slave #13
address 71: master #9
address 76: slave #9

**Reply by mwildbolz** [reply]
I think without being able to poll the bus (write to the bus) you already have the maximum possible.
For getting the groundwater temperatures you have to poll the HMU, as well for compressor data. This data is not sent to the bus without polling - as far as I have seen...

Sorry for this message :-(

**Reply by thoro** [reply]
:( - waiting for ebus adapter 3 then ...

**Comment by thoro** [comment]
Yes, I have the same issue, sent my USB Coupler to them but did not fix it :(

**Comment by edudele** [comment]


**Reply by thoro** [reply]
What devices do you have?

**Reply by edudele** [reply]
Esera USB adapter, I wrote it in the post.
Shelly EM Instant Power and Consumption.
Shelly 1 (temperature probe in radiators)

The demon gives me the rest of the data.

**Reply by thoro** [reply]
oh, sorry, I mean which Vaillant devices, i.e. which heatpump model

**Reply by edudele** [reply]
Hi,

Vaillant arotherm plus vwl 85/6
SensoConfort (vrc 720f)
Gateway VR921

**Reply by mwildbolz** [reply]
Using the VR921 gateway for polling the data is a real good idea. So you can get rid of the bus writing problem...
But the problem still existing is, that you cannot send messages to the heating system out of your home automation

**Reply by edudele** [reply]
I have not said that it is the solution, but in the face of scarcity, something is something

**Reply by Geg0r** [reply]
@edudele are you able to share your Grafana json file? Thx

**Reply by edsub** [reply]
> Using the VR921 gateway for polling the data is a real good idea. So you can get rid of the bus writing problem... But the problem still existing is, that you cannot send messages to the heating system out of your home automation

While I am waiting for my ordered Ebusd adapter 3 to be built I try to keep my sensoApp running so it will trigger (some more) of the output getting populated.
Based on that I pick up the available values via a custom python script from ebusd.
That works, but sometimes my phone kicks out the sensoApp and some values are not available (or do not get updated) anymore
Any suggestion on how to let the VR921 (without the sensoApp) trigger registers to provide data on its own?

**Comment by JeeCee1** [comment]
I have received my V3 adapter yesterday, when it's working I will sell my Esera, if anyone is interested.

**Comment by thoro** [comment]
Also got my V3 adapter and it's working flawlessly since then!

**Reply by RoDanny2021** [reply]
Hello, with which boiler do you use the adapter?

**Reply by thoro** [reply]
Heatpump - VWF117/4 with UniStor

**Reply by Geg0r** [reply]
@Thoro where are the values of the power consumption coming from? from the HMU or the BASV?

**Reply by thoro** [reply]
ebusd/hmu/CurrentConsumedPower

**Reply by Geg0r** [reply]
hmm okay this is empty for my VWF117/4 :-(
did you change the hmu.csv?

> root@ebusd:~# ebusctl info
> version: ebusd 23.1.23.1
> update check: OK
> device: 192.168.1.107:9999, enhanced, firmware 1.1[8419].2[847f]
> signal: acquired
> symbol rate: 24
> max symbol rate: 167
> min arbitration micros: 1
> max arbitration micros: 2
> min symbol latency: 7
> max symbol latency: 10
> reconnects: 0
> masters: 5
> messages: 509
> conditional: 0
> poll: 0
> update: 10
> address 03: master #11
> address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0308;HW=0403", loaded "vaillant/08.hmu.csv"
> address 10: master #2
> address 15: slave #2, scanned "MF=Vaillant;ID=72000;SW=0122;HW=7703", loaded "vaillant/15.720.csv"
> address 31: master #8, ebusd
> address 36: slave #8, ebusd
> address 71: master #9
> address 76: slave #9, scanned "MF=Vaillant;ID=VWZ00;SW=0308;HW=0403"
> address f1: master #10
> address f6: slave #10, scanned "MF=Vaillant;ID=NETX2;SW=4027;HW=5703"


> root@ebusd:~# ebusctl find
> general valuerange = no data stored
> hmu ConsumptionThisYear10 = no data stored
> hmu ConsumptionThisYear11 = no data stored
> hmu ConsumptionThisYear12 = no data stored
> hmu ConsumptionThisYear1 = no data stored
> hmu ConsumptionThisYear2 = no data stored
> hmu ConsumptionThisYear3 = no data stored
> hmu ConsumptionThisYear4 = no data stored
> hmu ConsumptionThisYear5 = no data stored
> hmu ConsumptionThisYear6 = no data stored
> hmu ConsumptionThisYear7 = no data stored
> hmu ConsumptionThisYear8 = no data stored
> hmu ConsumptionThisYear9 = no data stored
> hmu ConsumptionTotal = no data stored
> hmu CurrentCompressorUtil = no data stored
> hmu CurrentConsumedPower = no data stored
> hmu currenterror = -;-;-;-;-
> hmu CurrentYieldPower = no data stored
> hmu DateTime = valid;19:32:24;28.01.2023;0.625
> hmu EnergyCool = no data stored
> hmu EnergyHc = no data stored
> hmu errorhistory = no data stored
> hmu Hours = no data stored
> hmu HoursCool = no data stored
> hmu HoursHc = no data stored
> hmu SetMode = auto;29.0;-;-;0;1;1;0;0;0
> hmu State = no data stored
> hmu Status01 = 26.0;26.0;0.625;-;44.5;off
> hmu Status02 = no data stored
> hmu Status16 = no data stored
> hmu Status = no data stored
> hmu StatusCirPump = on
> hmu WaterThroughput = no data stored
> hmu YieldThisYear10 = no data stored
> hmu YieldThisYear11 = no data stored
> hmu YieldThisYear12 = no data stored
> hmu YieldThisYear1 = no data stored
> hmu YieldThisYear2 = no data stored
> hmu YieldThisYear3 = no data stored
> hmu YieldThisYear4 = no data stored
> hmu YieldThisYear5 = no data stored
> hmu YieldThisYear6 = no data stored
> hmu YieldThisYear7 = no data stored
> hmu YieldThisYear8 = no data stored
> hmu YieldThisYear9 = no data stored
> hmu YieldTotal = no data stored

**Reply by thoro** [reply]
Yes, I updated it to r1 - since it value needs to be read from hmu, doesn't come via read only - sorry

**Reply by Geg0r** [reply]
Would it be possible for you the share the csv file so that I can test it?

**Reply by thoro** [reply]
here is the extract for : 08.hmu.csv

```
r,,ch_18,,,,,18,,,UIN,,,,,,,,,,,,,,,,,,,,,
r,,ch_1e,,,,,1e,,,UIN,,,,,,,,,,,,,,,,,,,,,
r1,,FlowTempIn,,,,,1f,,,D2C,,,,,,,,,,,,,,,,,,,,,
r1,,FlowTempOut,,,,,20,,,D2C,,,,,,,,,,,,,,,,,,,,,
r1,,Integral,,,,,21,,,SIN,,,,,,,,,,,,,,,,,,,,,
r1,,SourceTempInput,,,,,22,,,D2C,,,,,,,,,,,,,,,,,,,,,
r1,,CurrentYieldPower,,,,,23,,,D1B,10,kW,aktuell gewonnene Energie aus der Umwelt in kW,,,,,,,,,,,,,,,,,,
r1,,CurrentConsumedPower,,,,,24,,,D1B,10,kW,aktueller Stromverbrauch der Wärmepumpe in kW,,,,,,,,,,,,,,,,,,
r1,,CurrentCompressorUtil,,,,,25,,,D2C,,%,aktuelle Kompressorauslastung in Prozent,,,,,,,,,,,,,,,,,,
r1,,SourceTempOutput,,,,,27,,,D2C,,,,,,,,,,,,,,,,,,,,,
r1,,FlowPressure,,,,,3d,,,D2C,4,,,,,,,,,,,,,,,,,,,,
r1,,SourcePressure,,,,,3e,,,D2C,4,,,,,,,,,,,,,,,,,,,,
r,,ch_3f,,,,,3f,,,UIN,,,,,,,,,,,,,,,,,,,,,
```

Just changed the r in the front to r1, that forces ebusd to read the value every  "cycle"

**Reply by Geg0r** [reply]
Cool thx this is working. 
Do you also know how to get not only CurrentConsumedPower for Heating but also for HotWater?
In this VRC 720 there is PrEnergySumHwcThisMontg but no current :-(

My goal is to see the current consumed power for heating, hot water and for the VWF itself..
any ideas?

Many thanks in advance!!

**Reply by thoro** [reply]
It doesn't exist, it's always heating and hot water. Total power consumption.

But I calculated it based on this:

```
sum_over_time((((1 - sum((circuit_status{device="hmu",sensor="heating_buffer_pump"}) > bool 0))) * on() sum(power{sensor="heatpump"}))[$__range:]) / on() (count_over_time(scrape_samples_scraped{job="stromzaehler"}[1m]) * 60) 
```

Basically if the ebusd/hmu/Status01/S5 - is 1 then the heating is active, otherwise it's hot water.

I also calculate the total heat:

```
sum_over_time((((water_throughput{device="hmu"} > 1300) * on () (temperature{sensor="heatpump",mode="vorlauf"} - on() temperature{sensor="heatpump",mode="rücklauf"}) * 1.16)/1000 * on() circuit_status{sensor="heating_buffer_pump"} ) [$__range:]) / on() (count_over_time(scrape_samples_scraped{job="stromzaehler"}[1m]) * 60)
```

Please ignore all the prometheus/grafana specific things, I map the MQTT results to prometheus.

**Comment by edsub** [comment]
How do I know I am suffering from this as well?

I think I have set the Esera potentiometer correctly.
But stuck now. Devices seem not to get recognized.

ebusctl info output:
```
version: ebusd 23.1.23.1-3-gaf614ae1
device: /dev/ttyUSB-EBUS
signal: acquired
symbol rate: 40
max symbol rate: 150
min arbitration micros: 118
max arbitration micros: 609
min symbol latency: 5
max symbol latency: 6
reconnects: 0
masters: 4
messages: 4
conditional: 0
poll: 0
update: 0
address 03: master #11
address 08: slave #11, scanning
address 10: master #2
address 31: master #8, ebusd
address 36: slave #8, ebusd, scanning
address 52: slave, scanning
address 71: master #9
address 76: slave #9, scanning
```

**Reply by edudele** [reply]
The essera device does not write to the bus.
so only your devices will only be recognized in the hello phase when they boot up.

you have two options.
1. Completely turn off the machine every time you start the daemon. (unfeasible)

2. disable scanconfig and generate a local repository only with the files that apply to your devices.

Read https://github.com/john30/ebusd-configuration/issues/270

**Reply by edsub** [reply]
Just read the thread you refer to.
At one moment this was my understanding as well, but at the same time I read about many others that do not seem to have an issue with the Esera USB adapter.
Next to that I saw that if my adapter sends (=write) something (eg ">31") it gets a response ("<31")  from the bus.
But nevermind, the behaviour I see matches your story.... :-(

Just ordered an ebusd adapter 3 and while I am waiting I guess I figure out which config files I need and work from there.......

Devices I have: (if you have knowledge on this, please let me know....

- Vaillant aroTHERM plus 75/6 monoblock heatpump =>;Vaillant;HMU00; ?
- Vaillant Unitower VIH QW 190/6 E with VWZ ai control unit => ;Vaillant;VWZ00; ?
- Vaillant VR70 module => ;Vaillant;VR_70; ?
- Vaillant sensoCOMFORT VRC 720 thermostat =>;Vaillant;CTLV2;
- Vaillant VR921 sensoNET gateway => ;Vaillant;NETX2;

**Reply by edsub** [reply]
I have these config files setup which seems to work for me
Config folder: _templates.csv  broadcast.csv  memory.csv  <vaillant folder>
vaillant folder: 08.hmu.csv  15.720.csv  52.vr_70.csv  _templates.csv  errors.inc  hcmode.inc  hwcmode.inc  iotesthp.inc  scan.csv
